src/include/os.h

```diff
@@ -139,6 +139,14 @@ struct __wt_file_handle_posix {
     int fd; /* POSIX file handle */
 
     bool direct_io; /* O_DIRECT configured */
+
+    /* The memory buffer and variables if we use mmap for I/O */
+    char *mmap_buf;
+    bool mmap_file_mappable;
+    int mmap_prot;
+    volatile uint32_t mmap_resizing;
+    size_t mmap_size;
+    volatile uint32_t mmap_usecount;
```

Comment:
These are used for synchronization. One thread can update them while the other needs to see the result of that update. So they need to be volatile to avoid compiler optimizations, where the thread waiting for an update keeps reading the value cached in the register and does not see the value updated in memory by another thread.

Reply:
Ummm, I looked explicitly for that and don't remember finding it. I'll take another look in the next review.